lib/pure/math.nim

```diff
@@ -941,6 +941,52 @@ func euclMod*[T: SomeNumber](x, y: T): T {.since: (1, 5, 1).} =
   if result < 0:
     result += abs(y)
 
+func ceilDiv*[T: SomeInteger](x, y: T): T {.inline, since: (1, 5, 1).} =
+  ## Ceil division is conceptually defined as `ceil(x / y)`.
+  ##
+  ## Assumes `x >= 0` and `y > 0` and (`x + y - 1 <= high(T)` if T is SomeUnsignedInt).
+  ##
+  ## This is different from the `system.div <system.html#div,int,int>`_
+  ## operator, which is defined as `trunc(x / y)`.
+  ## That is, `div` rounds towards `0` and `ceilDiv` rounds up.
+  ##
+  ## This function has above input limitation because it is implemented with
+  ## the expression that backend compiler generates faster code and it is
+  ## rarely used with negative value or unsigned int close to high(T)/2.
+  ## If you need `ceilDiv` works with any input, see:
+  ## https://github.com/demotomohiro/divmath
+  ##
+  ## **See also:**
+  ## * `system.div proc <system.html#div,int,int>`_ for integer division
+  ## * `floorDiv func <#floorDiv,T,T>`_ for integer division with rounds down.
+  runnableExamples:
+    assert ceilDiv( 12,  3) ==  4
+    assert ceilDiv( 13,  3) ==  5
+
+  when sizeof(T) == 8:
+    type UT = uint64
+  elif sizeof(T) == 4:
+    type UT = uint32
+  elif sizeof(T) == 2:
+    type UT = uint16
+  else:
+    type UT = uint8
+
+  assert x >= 0 and y > 0
+  when T is SomeUnsignedInt:
+    assert x + y - 1 >= x
+
+  # If divisor is const, backend compiler generate code without `div` instruction
+  # as it is slow on most of CPU.
+  # If divisor is a power of 2 and a const unsigned integer type,
+  # compiler generates faster code.
+  # If divisor is const and signed int, generated code become slower than
+  # the code with unsigned int because division with signed int need to works
+  # both positive and negative value without `idiv`/`sdiv`.
+  # That is why this code convert parameters to unsigned.
+  # And also this works with any positive int value unless T is unsigned.
```

Comment:
```suggestion
  # If divisor is const, the compiler generates code without a `div` instruction,
  # as it is slow on most CPUs.
  # If the divisor is a power of 2 and a const unsigned integer type, the
  # compiler generates faster code.
  # If the divisor is const and a signed integer, generated code become slower than
  # the code with unsigned integers, because division with signed integers need to works
  # for both positive and negative value without `idiv`/`sdiv`.
  # That is why this code convert parameters to unsigned.
  # And also this works with any positive integer value, unless T is unsigned.
```
Does it really matter that much, how much better the performance for const divisors is, anyway?

Reply:
Please see this post:
https://github.com/nim-lang/Nim/pull/18596#issuecomment-894420984

Reply:
That just explains how the performance for const divisors is faster afaict, not if it's really releveant in the first place.

Reply:
The compiler can calculate the return value at compile time only when all arguments are const.
There are cases where dividend is runtime value and divisor is const.
https://github.com/demotomohiro/ceil-division-in-wild